pennylane/ops/op_math/prod.py

```diff
@@ -186,6 +186,11 @@ def __copy__(self):
 
         return copied_op
 
+    # pylint: disable=protected-access
+    def _check_batching(self, params):
```

Comment:
For the composite operators, `Sum` and `Prod`, we want to be able to do `sum_op.data[i]` to access the data for the `i`th component. Otherwise it gets really tricky to figure out what parameter belongs to what component.